120-ng-manager/src/test/java/io/harness/ng/SecretManagerConnectorServiceImplTest.java

```
@@ -134,18 +139,24 @@ public void testCreateSecretManagerConnectorShouldFail_exceptionFromManager() th
   @Owner(developers = PHOENIKX)
   @Category(UnitTests.class)
   public void updateSecretManager() {
+    SecretRefData secretRefData = new SecretRefData(randomAlphabetic(10));
+    secretRefData.setDecryptedValue(randomAlphabetic(5).toCharArray());
     when(ngSecretManagerService.updateSecretManager(any(), any(), any(), any(), any()))
         .thenReturn(random(VaultConfigDTO.class));
     when(defaultConnectorService.update(any(), any())).thenReturn(null);
     when(connectorRepository.updateMultiple(any(), any())).thenReturn(null);
     when(defaultConnectorService.get(any(), any(), any(), any()))
-        .thenReturn(
-            Optional.of(ConnectorResponseDTO.builder()
-                            .connector(ConnectorInfoDTO.builder()
-                                           .connectorType(ConnectorType.VAULT)
-                                           .connectorConfig(VaultConnectorDTO.builder().isDefault(false).build())
-                                           .build())
-                            .build()));
+        .thenReturn(Optional.of(ConnectorResponseDTO.builder()
+                                    .connector(ConnectorInfoDTO.builder()
+                                                   .connectorType(ConnectorType.VAULT)
+                                                   .connectorConfig(VaultConnectorDTO.builder()
+                                                                        .isDefault(false)
+                                                                        .authToken(secretRefData)
+                                                                        .vaultUrl(randomAlphabetic(10))
+                                                                        .namespace(randomAlphabetic(5))
+                                                                        .build())
+                                                   .build())
+                                    .build()));
```

Can we have a test case for testing the mandatory fields i.e. If mandatory fields are not passed function should throw an error.